pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/upsertcompactmerge/UpsertCompactMergeTaskExecutor.java

```diff
@@ -99,10 +98,10 @@ protected List<SegmentConversionResult> convert(PinotTaskConfig pinotTaskConfig,
     // validate if partitionID is same for all small segments. Get partition id value for new segment.
     int partitionID = getCommonPartitionIDForSegments(segmentMetadataList);
 
-    // get the max creation time of the small segments. This will be the index creation time for the new segment.
-    Optional<Long> maxCreationTimeOfMergingSegments =
-        segmentMetadataList.stream().map(SegmentMetadataImpl::getIndexCreationTime).reduce(Long::max);
-    if (maxCreationTimeOfMergingSegments.isEmpty()) {
+    // get the max creation time from the task configuration passed by the generator
+    long maxCreationTimeOfMergingSegments =
+        Long.parseLong(configs.get(MinionConstants.UpsertCompactMergeTask.MAX_CREATION_TIME_MILLIS_KEY));
```

Comment:
This could be null during rollouts right?

Regardless, I think we should handle config.get returning null gracefully